src/main/java/uk/gov/justice/digital/domain/DomainExecutor.java

```diff
@@ -0,0 +1,271 @@
+    protected DataStorageService storage;
+    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(DomainExecutor.class);
+
+
+    public DomainExecutor(final String sourceRootPath, final String targetRootPath, final DomainDefinition domain,
+                          final DataStorageService storage) {
+        this.sourceRootPath = sourceRootPath;
+        this.targetRootPath = targetRootPath;
+        this.domainDefinition = domain;
+        this.storage = storage;
+    }
+
+    @Override
+    public void saveFull(final TableInfo info, final Dataset<Row> dataFrame, final String domainOperation)
+            throws DomainExecutorException {
+        logger.info("DomainOperations:: saveFull");
+        String tablePath = storage.getTablePath(info.getPrefix(), info.getSchema(), info.getTable());
+        if (domainOperation.equalsIgnoreCase("insert")) {
+            storage.create(tablePath, dataFrame);
+        } else if (domainOperation.equalsIgnoreCase("update")) {
+            if (storage.exists(info)) {
+                storage.replace(tablePath, dataFrame);
+            } else {
+                throw new DomainExecutorException("Delta table " + info.getTable() + "doesn't exists");
+            }
+        } else if (domainOperation.equalsIgnoreCase("sync")) {
+            if (storage.exists(info)) {
+                storage.reload(tablePath, dataFrame);
+            } else {
+                throw new DomainExecutorException("Delta table " + info.getTable() + "doesn't exists");
+            }
+        } else {
+            // Handle invalid operation type
+            logger.error("Invalid operation type " + domainOperation);
+            throw new DomainExecutorException("Invalid operation type " + domainOperation);
+        }
+        storage.endTableUpdates(info);
+        storage.vacuum(info);
+    }
+
+    @Override
+    protected void saveViolations(final TableInfo target, final Dataset<Row> dataFrame) {
+        String tablePath = storage.getTablePath(target.getPrefix(), target.getSchema(), target.getTable());
+        // save the violations to the specified location
+        storage.append(tablePath, dataFrame);
+        storage.endTableUpdates(target);
+    }
+
+    public void deleteFull(final TableInfo info) throws DomainExecutorException {
+        logger.info("DomainOperations:: deleteFull");
+        if (storage.exists(info)) {
+            storage.delete(info);
+            storage.endTableUpdates(info);
+            storage.vacuum(info);
+        } else {
+            throw new DomainExecutorException("Table " + info.getTable() + "doesn't exists");
+        }
+    }
+
+    public Dataset<Row> getAllSourcesForTable(final String sourcePath, final String source,
+                                              final TableTuple exclude) throws DomainExecutorException {
+        if(exclude != null && exclude.asString().equalsIgnoreCase(source)) {
+            //TODO: this condition only for unit test
```

Comment:
This can be removed since it isn't doing anything useful. You just need to invert the logic on the if condition.

Reply:
flipped the "if else" logic but we anyway need the else condition to print the table already present